odbc/src/api.rs

```diff
@@ -800,13 +801,41 @@ pub extern "C" fn SQLGetEnvAttr(
 
 #[no_mangle]
 pub extern "C" fn SQLGetEnvAttrW(
-    _environment_handle: HEnv,
-    _attribute: EnvironmentAttribute,
-    _value_ptr: Pointer,
+    environment_handle: HEnv,
+    attribute: EnvironmentAttribute,
+    value_ptr: Pointer,
     _buffer_length: Integer,
-    _string_length: *mut Integer,
-) -> SqlReturn {
-    unimplemented!()
+    string_length: *mut Integer,
+) -> SqlReturn {
+    let env_handle = MongoHandleRef::from(environment_handle);
+    env_handle.clear_diagnostics();
+    match env_handle.as_env() {
+        None => SqlReturn::INVALID_HANDLE,
+        Some(env) => {
+            let env_contents = env.read().unwrap();
+            unsafe {
+                *string_length = match value_ptr.is_null() {
+                    true => 0,
```

Comment:
Ah yes, I read this sentence a few times when figuring out what to set `string_length` to when `value_ptr` is null:

> If ValuePtr is NULL, StringLengthPtr will still return the total number of bytes (excluding the null-termination character for character data) available to return in the buffer pointed to by ValuePtr.

and wasn't sure if "the total number of bytes [...] available to return in the buffer" means the size of the buffer or the length of the attribute's value -- not sure how exactly to interpret "available to return" 😅 . I went with option one, but I can switch to option two if that's what you think they're saying.

> I tried the function with OdbcTest

I wasn't aware of ODBC Test until now! I'll try it out

> Even though the spec does not flag it as such, the DM takes care of HY092 (good since I did not see how this could be otherwise).

So we don't need to log a diagnostic record when an attribute value is invalid?